internal/controllers/gateway/gateway_controller.go

```diff
@@ -72,6 +74,12 @@ func (r *GatewayReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		return fmt.Errorf("publish service must be configured")
 	}
 
+	r.EnableReferenceGrant = ctrlutils.CRDExists(mgr.GetRESTMapper(), schema.GroupVersionResource{
```

Comment:
This - and similar field used in HTTPReconciler - will make the controller only enable the reference grant related checks when the CRD exists at the start of the controller right?

I believe we should explicitly state that in a code comment or elsewhere to ensure this is expected and not an implicit assumption.

Due to the "limitations) of controller-runtime I believe we can't easily work around this and dynamically change the behavior of Gateway or HTTP reconcilers. 🤔

Reply:
That's true. I made `enableReferenceGrant` private in both reconcilers to make it clear it's not expected to be set externally. Also added a comment explaining why we're resolving its value in `SetupWithManager`.

> Due to the "limitations) of controller-runtime I believe we can't easily work around this and dynamically change the behavior of Gateway or HTTP reconcilers. 🤔

Yeah, that would require introducing more layers, e.g. somehow making watches run by reconcilers to be dynamically set up once a CRD gets installed. I think this should not be a problem in reality given both experimental and standard Gateway API channels already have ReferenceGrants included.